cmd/sovereignnode/chainSimulator/common/common.go

```diff
@@ -2,10 +2,38 @@ package common
 
 import (
 	"github.com/multiversx/mx-chain-core-go/data"
+	"github.com/multiversx/mx-chain-go/config"
+	"github.com/multiversx/mx-chain-go/factory"
+	"github.com/multiversx/mx-chain-go/factory/runType"
 	"github.com/multiversx/mx-chain-go/node/chainSimulator/process"
+	sovRunType "github.com/multiversx/mx-chain-go/sovereignnode/runType"
 )
 
 // GetCurrentSovereignHeader returns current sovereign chain block handler from blockchain hook
 func GetCurrentSovereignHeader(nodeHandler process.NodeHandler) data.SovereignChainHeaderHandler {
 	return nodeHandler.GetChainHandler().GetCurrentBlockHeader().(data.SovereignChainHeaderHandler)
 }
+
+// CreateSovereignRunTypeComponents will create sovereign run type components
+func CreateSovereignRunTypeComponents(args runType.ArgsRunTypeComponents, sovereignExtraConfig config.SovereignConfig) (factory.RunTypeComponentsHolder, error) {
```

Comment:
The cleanest solution would be to move this function somewhere in `/sovereignnode/runtype` and use it in both chainSimualtor and sovereignNodeRunner

Reply:
Not quite same usage, just verified.
Will leave it as it is